### DISCUSSION TITLE
How to integrate the retrieval component with the answer generation component?

### DISCUSSION THREAD

**Discussion by FatemaD1577** [infiniflow/ragflow#4132] [Q&A]
### Describe your problem

I have created a simple rag flow for querying a knowledge base. Have configured the retrieval and generate component with the knowledge base input. But on querying getting error related to component id. Attaching a ss for reference: 

Retrieval Component:
![image](https://github.com/user-attachments/assets/c4d93b65-9919-4863-bf23-fab0676bce97)
![image](https://github.com/user-attachments/assets/f0465238-efdf-4947-9c1a-5332ace90639)

Generation Component:
![image](https://github.com/user-attachments/assets/597f9604-616a-46bc-ace6-23e1cd55acfc)
![image](https://github.com/user-attachments/assets/143b37bc-30c1-4052-9fcd-a014a6b4b638)

Flow:
![image](https://github.com/user-attachments/assets/6759bd25-03b6-41fd-9e1c-304276794a26)

Kindly let me know if I have missed out on something or configured incorrectly.How to integrate the retrieval component with the answer generation component?

**Comment by KevinHuSh** [comment]
You could link `Response_generation` back to `Human_assistant`.

**Reply by FatemaD1577** [reply]
Thanks for reverting back. How is the linking done? I have doubt in it as there is no information on that in the documentation.
